### DISCUSSION TITLE
Correct calculation for the displacement of the element with end releases.

### DISCUSSION THREAD

**Discussion by joshbushdash** [FEALiTE/FEALiTE2D#20] [General]
Thank you SaMohamed for your excellent work.

I have noticed an error in the post-processor where the displacement calculation of the element with end releases.

In lines 269 and 272 in PostProcessor, the displacement of the element should not be the full "dl" vector. Instead, the third and sixth angle values should be 0. Then the displacement value and diagram will be correct for elements with end releases.

Thank you for your attention to this matter~

**Comment by joshbushdash** [comment]
It looks like it is still not working...
Maybe the shape function for truss members should be established.

**Reply by joshbushdash** [reply]
So the question is that the rotation of the end release is unknown. Otherwise, will the shape function be correct if the rotation of the end release is known?

**Comment by SaMohamed** [comment]
This is a bit tricky and needs investigation. Do you have example for output from trusted solvers?
